src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Completion/TagHelperCompletionProvider.cs

```diff
@@ -212,11 +201,19 @@ static bool InOrAtEndOfAttribute(SyntaxNode attributeSyntax, int absoluteIndex)
         return completionItems;
     }
 
-    private static bool TryResolveInsertText(string baseInsertText, AttributeContext context, [NotNullWhen(true)] out string? snippetText)
+    private bool TryResolveInsertText(string baseInsertText, AttributeContext context, [NotNullWhen(true)] out string? snippetText)
     {
         if (context == AttributeContext.FullSnippet)
         {
-            snippetText = $"{baseInsertText}=\"$0\"";
+            if (_optionsMonitor.CurrentValue.AutoInsertAttributeQuotes)
+            {
+                snippetText = $"{baseInsertText}=\"$0\"";
+            }
+            else
+            {
+                snippetText = $"{baseInsertText}=$0";
```

Comment:
Do you know if the web tools option affects the `=` too, or literally just the quotes? I wonder about people who turn this off because they only want the editor to insert characters they type, and whether they'll still be annoyed when completing an item with Enter has it inserting an `=` they didn't type.

Reply:
For web tools it's slightly different because `=` is not a "commit character" in the sense that it's committing something from the completion list (this is my understanding from testing). We're using `=` as a commit character for completion as well. I could see the argument that we shouldn't... in which case we can change this. Having this in a few users hands and getting more feedback might be best. I can also @ a few of the vocal members from the issue and see if this behavior is what they expect.

My hunch, and based on the few feedbacks I looked at that were linked, people who type a whole thing and input `=` didn't want quotes added.